Feat: 개별작업 중이던 공통UI 이식 및 UI 테스트 코드 추가

gitgitWi opened the pull request:
## 프로젝트 전체 변경 사항

### `.gitignore`에 `yarn-error.log`, `coverage/` 등록

## 개별 변경사항

### 개별 Repo에서 작업중인 공통 UI 이식

- scss, store, router, types 포함

### UI 테스트 코드 추가

- 기본 Home 테스트 코드 추가
- 테스트 코드를 위한 설정 추가
  - ts-jest 설치
  - module-alias 설정 등

### 기타 변경사항

- `vue.d.ts` 파일 위치 이동, `packages/wiii` -> `packages/wiii/frontend`
  - 패키지 root에서는 동작 X
- frontend module-alias 설정 
- store 수정, mutationTypes -> actionTypes
  - mutation을 private하게 만들고 외부에선 actions만 사용
- event 타입 추가

## TODO

테스트 코드
- `router/routes` 파일에서 @ path 인식 안되는 문제, vue 파일 인식 안되는 문제 수정 필요
- click 이벤트 - 라우트 이동 구현 및 테스트 코드 수정

Commit:
Build: 일부 config 수정

- test coveratge 제외 파일 추가
- tsconfig.json에 root path alias 추가

Commit:
Fix: vue 타입선언 파일 위치 이동

- `vue.d.ts` 파일이 frontend 폴더 내 위치해야 정상 작동되는 것 확인

Commit:
Build: jest 사용을 위한 추가 설정

- TS 테스트를 위한 ts-jest 추가
- module alias를 frontend로 변경

Commit:
Feat: 개별작업 했던 기본 UI 이식

- 개별 Repo에서 만들었던 UI 이식
- scss, store, router, types 포함

elrion018 commented on packages/wiii/__tests__/Header.test.js:
```diff
@@ -0,0 +1,55 @@
+import Title from '@/components/header/Title';
+import MenuBar from '@/components/header/MenuBar';
+
+describe('Home view 마운트', () => {
```
테스트는 저도 적용해보면 좋을 것 같네요!

elrion018 commented on packages/wiii/frontend/components/header/index.vue:
```diff
@@ -0,0 +1,19 @@
+<template>
+  <header class="area">
+    <Title />
```
제 생각엔 Title 컴포넌트는 너무 코드 조각이 작은거 같아요.
혹시 이렇게 작게까지 나누신 이유가 있으신가요?

elrion018 commented on packages/wiii/frontend/components/header/Title.vue:
```diff
@@ -0,0 +1,18 @@
+    color: $blue-700;
+  }
+}
+</style>
```
Title 컴포넌트는 단독으로 나누기에 너무 작은거 같아요!
그리고 ZUM 인베스팅과 같은 상수는 따로 관리하는 것이 어떨까요?

elrion018 commented on packages/wiii/frontend/components/header/Title.vue:
```diff
@@ -0,0 +1,18 @@
+<template>
+  <div class="area" id="header-title">
+    <h1>ZUM 인베스팅</h1>
+  </div>
+</template>
+
+<script>
+export default {};
+</script>
+
+<style lang="scss" scoped>
+#header-title {
+  h1 {
+    @extend .card;
+    color: $blue-700;
+  }
+}
+</style>
```
Title 컴포넌트는 단독으로 나누기에 너무 작은거 같아요!
그리고 'ZUM 인베스팅'과 같은 상수는 따로 관리하는 것이 어떨까요?

gitgitWi replied:
Title 은 굳이 안나눠도 될것 같긴하네요 ㅋㅋ

타이틀 문구는 테스트에서도 사용해야 하니
types에 저장하면 좀더 관리하기 편할 것 같습니다!

dogyeong commented on packages/wiii/frontend/components/header/index.vue:
```diff
@@ -0,0 +1,19 @@
+<template>
+  <header class="area">
+    <Title />
+    <menu-bar />
```
`menu-bar`도 케이스를 통일시켜주는게 좋을 것 같습니다~ `MenuBar`

gitgitWi replied:
앗..감사합니다 ㅋㅋ

dogyeong reviewed (approved):
수고하셨습니다 👍

Commit:
Test: MenuIcons 테스트 추가

- MenuIcons 테스트 추가
- store 수정, mutationTypes -> actionTypes
  - mutation을 private하게 만들고 외부에선 actions만 사용
- event 타입 추가

## TODO
- routes 파일에서 @ path 인식 안되는 문제, vue 파일 인식 안되는 문제 수정 필요
- click 이벤트 - 라우트 이동 구현 및 테스트 코드 수정

gitgitWi commented:
![image](https://user-images.githubusercontent.com/57997672/118257696-e490b300-b4e9-11eb-9ee5-dbb3937a3a56.png)

메뉴 `click` 이벤트 - 라우팅 테스트 실패 상태

elrion018 commented:
수고하셨습니다!
